src/phantom.js

```diff
@@ -37,6 +37,7 @@ const defaultLogger = createLogger();
  */
 export default class Phantom {
     logger: Logger;
+    doing_NOOP: boolean;
```

Comment:
Because your heartbeat is on a setInterval, you could end up sending a NOOP while one is still in transit / been processed.  This could cause a backlog of NOOP request's, in effect causing a memory leak.  The PhantomJS end send's an acknowledgement when it's done with the NOOP.